buf/registry/plugin/v1/plugin.proto

```diff
@@ -0,0 +1,130 @@
+  ];
+  // The Plugin's state, either active or deprecated.
+  PluginState state = 7 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The configurable description of the Plugin.
+  string description = 8 [(buf.validate.field).string.max_len = 350];
+  // The configurable URL in the description of the Plugin.
+  string url = 9 [
+    (buf.validate.field).string.uri = true,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
+  ];
+  // Semver-formatted release version of the Plugin.
+  string version = 10;
+  // The revision of the Plugin. This represents the BSR image version of the Plugin.
+  uint32 revision = 11;
+  // The full container image digest associated with this revision.
+  // Ref: https://github.com/opencontainers/image-spec/blob/main/descriptor.md#digests
+  string container_image_digest = 12;
+}
+
+// The visibility of a Plugin, currently either public or private.
+enum PluginVisibility {
+  PLUGIN_VISIBILITY_UNSPECIFIED = 0;
+  // PLUGIN_VISIBILITY_PUBLIC says that the Plugin is publicly available.
+  PLUGIN_VISIBILITY_PUBLIC = 1;
+  // PLUGIN_VISIBILITY_PRIVATE says that the Plugin is private.
+  PLUGIN_VISIBILITY_PRIVATE = 2;
+}
+
+// The state of a Plugin, currently either active or deprecated.
+enum PluginState {
+  PLUGIN_STATE_UNSPECIFIED = 0;
+  // PLUGIN_STATE_ACTIVE says that the Plugin is currently active.
+  PLUGIN_STATE_ACTIVE = 1;
+  // PLUGIN_STATE_DEPRECATED says that the Plugin has been deprecated and should not longer be
+  // used.
+  PLUGIN_STATE_DEPRECATED = 2;
+}
+
+// PluginRef is a reference to a Plugin, either an id or a fully-qualified name and its version and revision details.
+//
+// This is used in requests.
+message PluginRef {
+  option (buf.registry.priv.extension.v1beta1.message).request_only = true;
+
+  // The fully-qualified name of a Plugin within a BSR instance.
+  //
+  // In combination with its version and revision, a Name uniquely identifies a Plugin.
+  // This is used for requests when a caller only has the name and version components of the Plugin and not the ID.
+  message Name {
+    // The name of the owner of the Plugin, either a User or Organization.
+    string owner = 1 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.max_len = 32
+    ];
+    // The name of the Plugin.
+    string plugin = 2 [(buf.validate.field).string = {
+      min_len: 2
+      max_len: 100
+    }];
+  }
+
+  oneof value {
+    option (buf.validate.oneof).required = true;
+    // The id of the Plugin.
+    string id = 1 [(buf.validate.field).string.tuuid = true];
+    // The fully-qualified name of the Plugin.
+    Name name = 2;
+  }
+  // Semver-formatted release version of the Plugin.
+  string version = 3;
+  // The revision of the Plugin. This represents the BSR image version of the Plugin.
+  uint32 revision = 4;
```

Comment:
Should these be under `Name`? Why would I want to specify these if I have the plugin ID?